libraries/CMakeLists.txt

```diff
@@ -1,3 +1,16 @@
+if( MINGW )
```

Comment:
I find it confusing to have CURL_LIBRARIES and CURL_ADDITIONAL_LIBS, and to have them detected / modified / used differently in different places. Perhaps unify and move this into top-level CMakeLists.txt?